src/lib.rs

```diff
@@ -143,7 +154,6 @@
 //! [`no_std` environment]: https://docs.rust-embedded.org/book/intro/no-std.html
 
 #![cfg_attr(not(feature = "use_std"), no_std)]
-
```

Comment:
It looks like you're using an IDE or some tool to automatically format code. Could you please revert all styling changes, so that the diff can be shrunk down to contain only the actual logic changes?

Reply:
I could separate stylistic changes in a separate commit. Does you project rely on rustfmt?

It's possible to add `.rustfmt.yml` configuration to tune the formatter behavior.

Reply:
I've isolated stylistic changes in a separate commit.

Reply:
This project currently does not use rustfmt, although this could change in the future. The current stable version of rustfmt does not support all of the styling I want for this crate. The unstable version has some features I like, but until those are stabilized, I would like to avoid using rustfmt

Reply:
Excluded formatting commit